pkg/registry/batch/job/strategy.go

```diff
@@ -236,8 +243,11 @@ func generateSelector(obj *batch.Job) {
 	if obj.Spec.Selector.MatchLabels == nil {
 		obj.Spec.Selector.MatchLabels = make(map[string]string)
 	}
-	if _, found := obj.Spec.Selector.MatchLabels["controller-uid"]; !found {
-		obj.Spec.Selector.MatchLabels["controller-uid"] = string(obj.ObjectMeta.UID)
+
+	for _, value := range []string{"controller-uid"} {
```

Comment:
Don't change this, yet, but add here a TODO, that we'll want to change that in the future.

Reply:
why can't we change this to use just the new label?

Anyone that is relying on this selector (for example, the job controller itself) should be able to use it without caring what the labels are.

Reply:
@sftim brought up a possible bug if there is an upgrade.

1.27 behavior:

a) Legacy labels are present
b) Legacy labels (controller-uid) is used as selector.

Upgrade of Kubernetes (1.28)

New labels would be added for new jobs and the new label would be used for selector.  

I don't really know if upgrades would change the selector and I was being on the conservative side so we can make sure that both labels are present so we could switch to use new selector once the upgrade paths that we can support have both labels.

Reply:
> New labels would be added for new jobs and the new label would be used for selector.

Exactly, already existing jobs wouldn't change, as this is part of the creation strategy, not update.

Reply:
https://github.com/kubernetes/kubernetes/issues/113714#issuecomment-1400581581 was the discussion that led me to implement just using the old label selector.

So you are saying that upgrades should keep the old labels and I don't think we allow updates to the LabelSelector?

Reply:
I mistakenly assumed that selector was hardcoded in the controller, but it's not.